### PR TITLE
feat: pict-rs v0.5.4 & Lemmy 0.19.3

### PR DESCRIPTION
**Pull request by ticoombs**
Closes: #215 #208

### CODE REVIEW

**Comment by dessalines** [comment]
It'd also be nice if there were lemmy-ansible / docker specific 0.4 -> 0.5 upgrade instructions on the readme.

**Comment by ticoombs** [comment]
@dessalines That's why it's still in a draft state ;)

**Comment by dessalines** [comment]
@asonix Could you possibly give some lemmy-specific pictrs v0.4-v0.5 upgrade instructions?

I see this upgrade guide: https://git.asonix.dog/asonix/pict-rs/#0-4-to-0-5-migration-guide

But it doesn't say how to perform it *offline*, or for docker. @Nutomic is also saying that an intermediate step was necessary to go to `0.4.7` **first**, then to `0.5` . We need these upgrade instructions fully detailed.

**Comment by Nutomic** [comment]
you can either run 0.4.7 and then call POST /internal/prepare_upgrade
its documented here: https://git.asonix.dog/asonix/pict-rs/src/tag/v0.4.7
or go directly to 0.5 without preparing, then the db migration will be performed on first start which means a bit of downtime

**Comment by ticoombs** [comment]
> you can either run 0.4.7 and then call POST /internal/prepare_upgrade
Which we have already done as part of 1.3.1 and I am going to assume everyone that uses the ansible repo has done. 

The only items to really check are for the new 0.5 environment variables, and provide those correct variables under example vars, and possibly a check (testing on our side) to confirm if wrong variables in the pictrs docker will block it from starting.

**Comment by Nutomic** [comment]
Config changes are documented here: https://git.asonix.dog/asonix/pict-rs/#user-content-configuration-updates

File: README.md
Comment (dessalines) [inline README.md:103]: Amend

**Review by dessalines — COMMENTED** [review]
@Nutomic If you would, please add command-line instructions for the no-downtime upgrade path as well.

File: README.md
Comment (Nutomic) [inline README.md:105]: Also mention about running `POST /internal/prepare_upgrade` with 0.4 to reduce downtime with the migration. Its documented [here](https://git.asonix.dog/asonix/pict-rs/src/branch/v0.4.x), Ctrl+F for `prepare_upgrade`.

**Review by codyro — APPROVED** [review]

File: README.md
Comment (codyro) [reply README.md:106]: It was added as an optional note but the conversation didn't get marked as resolved. I think this should be sufficient, unless you're looking for a more prominent mention for larger instances.

https://github.com/LemmyNet/lemmy-ansible/pull/213/commits/1c33460ffcd6f88191692ce0ccec057f96712a0a

**Comment by ticoombs** [comment]
Will wait for @Nutomic / @dessalines to also Approve before merging after reading: https://github.com/LemmyNet/lemmy-ansible/pull/213/commits/1c33460ffcd6f88191692ce0ccec057f96712a0a

**Review by Nutomic — APPROVED** [review]

**Comment by ticoombs** [comment]
:rocket: Merging

**Pull request merged**
